tests/Tests.hs

```diff
@@ -65,15 +90,41 @@ killOpenWrite = do
     _ <- S.seccomp_rule_add_array ctx S.SCMP_ACT_KILL S.SCopen [S.ArgCmp 1 S.MASQUED_EQ 0x3 0x1]
     _ <- S.seccomp_load ctx
     S.seccomp_release ctx
-    _ <- IO.openFd "/dev/null" IO.WriteOnly Nothing IO.defaultFileFlags
+    _ <- PosixIO.openFd "/dev/null" PosixIO.WriteOnly Nothing PosixIO.defaultFileFlags
+    return ()
+
+actErrno :: Assertion
+actErrno = do
+    ErrNo.resetErrno
+    -- trying to capture the syscall that should fail with an errno we can test
+    ctx <- S.seccomp_init (S.SCMP_ACT_ERRNO 42)
+    whitelistHaskellRuntimeCalls ctx
+    _ <- S.seccomp_load ctx
+    S.seccomp_release ctx
+    errno <- ErrNo.getErrno
+    when (errno /= ErrNo.eOK) exitFailure
+    -- triggering prohibited action
+    _ <- Control.Exception.catch (System.IO.openFile "/dev/null" System.IO.ReadMode) $ \e -> do
+            putStrLn ("caugth error: " ++ show (e::Control.Exception.SomeException))
+            ErrNo.Errno errno <- ErrNo.getErrno
+            if (errno == 42)
+            then exitSuccess
+            else (do
+                    putStrLn ("unexpected errno: " ++ show errno)
+                    exitFailure
+                  )
+            -- return dummy handle to make it compile. not reached
+            return System.IO.stdin
+    putStrLn "unreachable"
+    exitFailure
     return ()
 
 
 assertNotTerminated :: IO () -> Assertion
 assertNotTerminated f = do
     pid <- forkProcess f
     result <- getProcessStatus True False pid
-    assertBool ("unexpected terminate: " ++ show result) (result /= Just (Terminated sigSYS False))
+    assertBool ("unexpected terminate: " ++ show result) (result == Just (Exited ExitSuccess))
```

Comment:
Perhaps rename `assertNotTerminated` to `assertExitSuccess` (because we're relying on those semantics in actErrno)?